### ISSUE TITLE
Reduce package dependencies

### ISSUE THREAD

**Issue by hadjipantelis** [functionaldata/tPACE#28]
For example:  Within `fitted.FPCA`, `Hmisc::approxExtrap(...)` can probably be dropped in favour of `stats::spline( method = "natural", ... )`. The performance deterioration will probably be minimal but we will be able to drop the depend to `Hmisc` which is a big dependency. Similarly dependencies on `pracma` and/or `MASS` probably could be trimmed too.

**Comment by CrossD** [comment]
I agree with dropping redundant dependencies. We may port `Hmisc::approxExtrap` to `fdapace` instead of using spline smoothing for the sake of minimizing changes. `pracma` may also be dropped, but I think since everyone has `MASS` installed it should be less of concern.

**Comment by hadjipantelis** [comment]
Agreed. I am planning to make the replacement, check if we get too many unit-tests complaining and then potentially re-write if needed.

**Comment by ChingChuan-Chen** [comment]
I was working on matlab function interp in R. 
I can contribute that code which is written in RcppArmadillo and verified. I think that it’s what you need. 

You can find interp.cpp in 
[url](https://drive.google.com/drive/folders/1hgqpup6vyljhgxWM92scceXmPVsCKToE).

**Comment by hadjipantelis** [comment]
Hello ChingChuan, thanks for being interested to contribute to `fdapace`. `fdapace` doesn't depend on ‘RccArmadillo` so there is a bit of an issue of introducing a new dependency and potentially not dropping a single one. You are welcome to fork out `fdapace` and integrate your code, we will be happy to review the relevant pull request.
